LanguageFeatures/Patterns/execution_pattern_for_statement_A01_t05.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// A statement of the form:
+///
+/// for (<patternVariableDeclaration>; <condition>; <increment>) <statement>
+///
+/// Is executed like a traditional for loop though is more likely to declare
+/// multiple variables.
+///
+/// @description Check that pattern-for statement is executed like a traditional
+/// for loop. Test object pattern
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns,records
+
+import "../../Utils/expect.dart";
+import "patterns_lib.dart";
+
+main() {
+  String log = "";
+  for (var Square(:areaAsInt) = Square(1); areaAsInt <= 9;
+      Square(:areaAsInt) = Square(++areaAsInt)) {
```

Comment:
Again, I'd expect log entries "1;4" because the third square would be `Square(5)`, not `Square(3)`. Perhaps `areaAsInt <= 25` would capture the intended behavior just fine?